tests/unit/utils/test_file_utils.py

```diff
@@ -0,0 +1,45 @@
+from griptape.loaders import TextLoader
+from griptape import utils
+from tests.mocks.mock_embedding_driver import MockEmbeddingDriver
+
+MAX_TOKENS = 50
+
+
+class TestFileUtils:
+    def test_load_file(self):
+        file = utils.load_file("tests/resources/test.txt")
+
+        assert file.decode("utf-8").startswith("foobar foobar foobar")
+        assert len(file.decode("utf-8")) == 4563
+
+    def test_load_files(self):
+        sources = ["tests/resources/test.txt", "tests/resources/test.txt", "tests/resources/small.png"]
+        files = utils.load_files(sources)
+        assert len(files) == 2
+
+        test_file = files[utils.str_to_hash("tests/resources/test.txt")]
+        assert len(test_file) == 4563
+        assert test_file.decode("utf-8").startswith("foobar foobar foobar")
+
+        small_file = files[utils.str_to_hash("tests/resources/small.png")]
+        assert len(small_file) == 97
+        assert small_file[:8] == b"\x89PNG\r\n\x1a\n"
+
+    def test_load_file_with_loader(self):
+        file = utils.load_file("tests/resources/test.txt")
+        artifacts = TextLoader(max_tokens=MAX_TOKENS, embedding_driver=MockEmbeddingDriver()).load(file)
+
+        assert len(artifacts) == 39
+        assert isinstance(artifacts, list)
+        assert artifacts[0].value.startswith("foobar foobar foobar")
+
+    def test_load_files_with_loader(self):
+        sources = ["tests/resources/test.txt"]
+        files = utils.load_files(sources)
+        loader = TextLoader(max_tokens=MAX_TOKENS, embedding_driver=MockEmbeddingDriver())
+        collection = loader.load_collection(list(files.values()))
```

Comment:
Wondering if `load_files` should just return a list so it's easier to pass into `load_collection`? Don't think we can load the files concurrently if we change that though.

Reply:
You could have two functions, something like  `load_file_collection` and `load_file_list`.  Then you can identify individual files in the response if you need to, otherwise the `load_file_list` will be more convenient. The implementation of `load_file_list` could just call `load_file_collection` (obvisously).

Reply:
Another more heavy handed alternative could be to allow passing a `dict[str, bytes]` into load_collection in addition to `list[bytes]` for each of the relevant loaders. A pro to this option is that you could even use the original key passed in to the input, so the client needed to map from a file name to the file contents, they wouldn't need to do two lookups.

Currently:
```
sources = [ 'foo.txt' ]
content_by_filename_hash = utils.load_files(sources)
artifacts_by_content_hash = loader.load_collection(list(files_by_filename.values()))

foo_content = content_by_filename_hash[hash_from_str('foo.txt')]
foo_artifact = artifacts_by_content_hash[loader.to_key(foo_content)]

```

With suggestion:
```
sources = [ 'foo.txt' ]
content_by_filename_hash = utils.load_files(sources)
artifacts_by_filename_hash = loader.load_collection(content_by_filename_hash)

# One less lookup if `load_collection` reuses keys when passed a dict
foo_artifact = artifacts_by_filename_hash[hash_from_str('foo.txt')]
```

If the input is the same shape as the output of `load_collection`, then it'd make it easier to chain together loaders in general, however I can't think of a use case besides this one.

---

> Another alternative that I'm not necessarily advocating for due to increase in scope, but we could make loaders composable and re-introduce file loader.